### ISSUE TITLE
The talker_hidden_states not obtained correctly

### ISSUE THREAD

**Issue by howitry** [QwenLM/Qwen3-TTS#337]
### Description

Error code location: 

### Reproduction

1. Error code location: 
  https://github.com/QwenLM/Qwen3-TTS/blob/main/finetuning/sft_12hz.py#L108
2. Error Description：
  The position where codec_mask is True corresponds to the actual codec. hidden_states[codec_mask[:, :-1]] retrieves the hidden_states of the actual codec. Note that the hidden state at time t is used to predict the codec at time t+1.
  The above logic cannot be matched with the code: https://github.com/QwenLM/Qwen3-TTS/blob/main/qwen_tts/core/models/modeling_qwen3_tts.py#L1612
  The hidden state at time t should not be concatenated with the codec embedding at time t (the hidden state at time t-1 is used to predict the codec at time t), but should be concatenated with the codec embedding at time t+1.
  The correct value is `talker_hidden_states=hidden_states[codec_mask[:, 1:]]`, https://github.com/QwenLM/Qwen3-TTS/blob/main/finetuning/sft_12hz.py#L108

### Logs

```shell

```

### Environment Information

no Environment  

### Known Issue

- [ ] The issue hasn't been already addressed in Documentation, Issues, and Discussions.

**Comment by wincing2** [comment]
Did you notice any improvement in model performance after your update?

**Comment by Zahgrom34** [comment]
Yes, huge improvement, but only combined with the loss shift fix from #278. The two bugs stack: the main loss trains t+2 instead of t+1, and the sub-talker reads the residual codes it's supposed to predict straight out of its own input (they are embedded into the hidden state at that position, so it's label leakage, at inference that shortcut doesn't exist).

With both fixed, same dataset that previously produced distorted noise gave natural sounding speech in one run. Fixing only one of them is not enough, I tried.

Equivalent fix to yours that I used, in case indexing style matters to anyone:

```python
prev_mask = codec_mask.roll(-1, 1)
prev_mask[:, -1] = False
talker_hidden_states = hidden_states[prev_mask]
```

**Comment by wincing2** [comment]
What's the learning rate that you use? I remember setting the learning rate too high causes the output of noise.

**Comment by Zahgrom34** [comment]
The lr was 2e-5 in all runs, same value before and after the fix, so it's not that

**Comment by Zahgrom34** [comment]
the neat part is you can prove it's not an lr thing without training at all, take the stock model, generate some audio, then teacher force its own generated codes back through the training loss. A correct objective should give near zero CE on the model's own output.

with the loss path as shipped, manual shift + HF's internal shift = double shift we get CE 9.07, which is worse than uniform random over the codebook ln(2048) = 7.62.   with the single shift it's 0.55, as expected

**Comment by Zahgrom34** [comment]
So with the shipped objective the model is literally being trained to predict code t+2 while inference feeds back under the t+1 contract, any lr converges to that wrong mapping, low lr just gets you there slower. That's why the loss curve looks totally healthy while the audio is noise

**Comment by wincing2** [comment]
> The lr was 2e-5 in all runs, same value before and after the fix, so it's not that

Did you use this lr on 0.6B or 1.7B model?

**Comment by Zahgrom34** [comment]
Mine was 1.7B, bf16, on a single L40S. Same 2e-5 on it before and after the fix, noise before, much more clear and reasonable speech after, so lr wasn't the only thing
